File: Patterns/Adapter Pattern/CSharp/DuckTestDrive.cs
using System;

using Duck;
using Turkey;

public class DuckTestDrive
{
    public static void TestDuck(Duck.Duck duck)
    {
        duck.Quack();
        duck.Fly();
    }

    public static void Main(string[] args)
    {
        Duck.Duck duck = new Duck.MallardDuck();

        Turkey.Turkey turkey = new Turkey.WildTurkey();
        Duck.Duck turkeyAdapter = new Turkey.TurkeyAdapter(turkey);

        Console.WriteLine("칠면조가 말하길");
        turkey.Gobble();
        turkey.Fly();

        Console.WriteLine("\n오리가 말하길");
        TestDuck(duck);

        Console.WriteLine("\n칠면조 어뎁터가 말하길");
        TestDuck(turkeyAdapter);
    }
}

File: Patterns/Adapter Pattern/Cpp/DuckTestDrive.cpp
#include <iostream>

#include "./Duck/MallardDuck.cpp"
#include "./Turkey/WildTurkey.cpp"
#include "./Turkey/TurkeyAdapter.cpp"

using namespace std;

void testDuck(Duck* duck) {
    duck->quack();
    duck->fly();
}

int main() {
    Duck* duck = new MallardDuck();

    Turkey* turkey = new WildTurkey();
    Duck* turkeyAdapter = new TurkeyAdapter(turkey);

    cout << "칠면조가 말하길\n";
    turkey->gobble();
    turkey->fly();

    cout << "\n오리가 말하길\n";
    testDuck(duck);

    cout << "\n칠면조 어뎁터가 말하길\n";
    testDuck(turkeyAdapter);
}